### PR TITLE
forge: begin converting a few smoke-test tests

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

File: testsuite/smoke-test/src/event_fetcher.rs [hunk 1]
```diff
@@ -1,63 +1,85 @@
...
+        client.submit(&txn)?;
+        client.wait_for_signed_transaction(&txn, None, None)?;
+
+        let txn = account1.sign_with_transaction_builder(factory.peer_to_peer(
+            Currency::XUS,
+            account2.address(),
+            4_000_000,
+        ));
+        client.submit(&txn)?;
+        client.wait_for_signed_transaction(&txn, None, None)?;
```
Comment (davidiw) [inline testsuite/smoke-test/src/event_fetcher.rs:47]: why not make a submit and wait?

File: testsuite/smoke-test/src/event_fetcher.rs [hunk 2]
```diff
@@ -1,63 +1,85 @@
...
+
+        let events_fetcher = DiemEventsFetcher::new(ctx.url())?;
```
Comment (davidiw) [inline testsuite/smoke-test/src/event_fetcher.rs:49]: considering this is only used in the runtime, why not instantiate it there? I guess the ctx?

File: testsuite/smoke-test/src/transaction.rs [hunk 3]
```diff
@@ -1,126 +1,114 @@
...
+        match txn.transaction {
+            TransactionDataView::UserTransaction {
+                sender,
+                sequence_number,
+                max_gas_amount,
+                gas_unit_price,
+                gas_currency,
+                script,
+                ..
+            } => {
```
Comment (davidiw) [inline testsuite/smoke-test/src/transaction.rs:85]: my only thoughts are an if let would be one less indentation ;)

**Review by davidiw — APPROVED** [review]
overall seems good, couple of nits, but it is more about the framework and not this code.

File: testsuite/smoke-test/src/transaction.rs (hunk 3 above)
Comment (bmwill) [reply testsuite/smoke-test/src/transaction.rs:85]: This is code that existed previously, I tried to make as few changes to the existing tests as possible. But yeah I agree :)

**Comment by davidiw** [comment]
You could at least humor me by answering my questions before rushing to land this sucker :P

File: testsuite/smoke-test/src/event_fetcher.rs (hunk 2 above)
Comment (bmwill) [reply testsuite/smoke-test/src/event_fetcher.rs:49]: Not sure I follow what you're asking? What runtime? This is a construct that is only being tested and used in this single test.

File: testsuite/smoke-test/src/event_fetcher.rs (hunk 1 above)
Comment (bmwill) [reply testsuite/smoke-test/src/event_fetcher.rs:47]: Yeah that would be a good helper method to add to the clients

**Comment by bmwill** [comment]
@davidiw I answered all of your questions?

**Pull request merged**